# rosalind

Learning bioinformatics usually requires solving computational problems of varying difficulty that are extracted from real challenges of molecular biology.

To make learning bioinformatics fun and easy, we have founded Rosalind, a platform for learning bioinformatics through problem solving.

Rosalind offers an array of intellectually stimulating problems that grow in biological and computational complexity; each problem is checked automatically, so that the only resource required to learn bioinformatics is an internet connection.

Rosalind also promises to facilitate improvements in standard bioinformatics education by providing a vital teaching aid and a central homework resource.

Rosalind is inspired by Project Euler, Google Code Jam, and the ever growing movement of free online courses. The project's name commemorates Rosalind Franklin, whose X-ray crystallography with Raymond Gosling facilitated the discovery of the DNA double helix by Watson and Crick.

### http://rosalind.info/users/rosli/

## http://rosalind.info/problems/list-view/?location=algorithmic-heights

    ID	    / Title
1)  FIBO	/ Fibonacci Numbers  :heavy_check_mark:
2)  BINS	/ Binary Search  (//)
3)  DEG	    / Degree Array  (//)
4)  INS	    / Insertion Sort  (//)
5)  DDEG	/ Double-Degree Array  (//)
6)  MAJ	    / Majority Element  (//)
7)  MER	    / Merge Two Sorted Arrays  (//)
8)  2SUM	/ 2SUM   (//)
9)  BFS	    / Breadth-First Search  (//)
10) CC	    / Connected Components  (//)
11) HEA	    / Building a Heap  (//)
12) MS	    / Merge Sort    (//)
13) PAR	    / 2-Way Partition
14) 3SUM	/ 3SUM  (//)
15) BIP	    / Testing Bipartiteness  (//)
16) DAG	    / Testing Acyclicity  (//)
17) DIJ	    / Dijkstra's Algorithm  (//)
18) HS	    / Heap Sort (//)
19) INV	    / Counting Inversions
20) PAR3	/ 3-Way Partition
21) SQ	    / Square in a Graph
22) BF	    / Bellman-Ford Algorithm  (//)
23) CTE	    / Shortest Cycle Through a Given Edge
24) MED	    / Median  (//)
25) PS	    / Partial Sort  (//)
26) TS	    / Topological Sorting
27) HDAG	/ Hamiltonian Path in DAG
28) NWC	    / Negative Weight Cycle
29) QS	    / Quick Sort
30) SCC	    / Strongly Connected Components
31) 2SAT	/ 2-Satisfiability
32) GS	    / General Sink
33) SC	    / Semi-Connected Graph
34) SDAG	/ Shortest Paths in DAG

## http://rosalind.info/problems/list-view/?location=bioinformatics-stronghold

    ID      / Title   
1) DNA      / Counting DNA Nucleotides
2) RNA      / Transcribing DNA into RNA
3) REVC     / Complementing a Strand of DNA
4) FIB      / Rabbits and Recurrence Relations
5) GC       / Computing GC Content
6) HAMM     / Counting Point Mutations
7) PROT     / Translating RNA into Protein
8) SUBS     / Finding a Motif in DNA
9) IPRB     / Mendel's First Law
10) CONS    / Consensus and Profile
11) FIBD    / Mortal Fibonacci Rabbits
12) GRPH    / Overlap Graphs
13) LCSM    / Finding a Shared Motif
14) MPRT    / Finding a Protein Motif
15) MRNA    / Inferring mRNA from Protein
16) ORF     / Open Reading Frames
17) PERM    / Enumerating Gene Orders
18) PRTM    / Calculating Protein Mass
19) REVP    / Locating Restriction Sites
20) SPLC    / RNA Splicing
21) IEV     / Calculating Expected Offspring
22) LIA     / Independent Alleles
23) LEXF    / Enumerating k-mers Lexicographically
24) LGIS    / Longest Increasing Subsequence
25) LONG    / Genome Assembly as Shortest Superstring
26) PMCH    / Perfect Matchings and RNA Secondary Structures
27) PPER    / Partial Permutations
28) PROB    / Introduction to Random Strings
29) SIGN    / Enumerating Oriented Gene Orderings
30) SSEQ    / Finding a Spliced Motif
31) TRAN    / Transitions and Transversions
32) TREE    / Completing a Tree
33) CAT     / Catalan Numbers and RNA Secondary Structures
34) CORR    / Error Correction in Reads
35) INOD    / Counting Phylogenetic Ancestors
36) KMER    / k-Mer Composition
37) KMP     / Speeding Up Motif Finding
38) LCSQ    / Finding a Shared Spliced Motif
39) LEXV    / Ordering Strings of Varying Length Lexicographically
40) MMCH    / Maximum Matchings and RNA Secondary Structures
41) PDST    / Creating a Distance Matrix
42) REAR    / Reversal Distance
43) RSTR    / Matching Random Motifs
44) SSET    / Counting Subsets
45) ASPC    / Introduction to Alternative Splicing
46) EDIT    / Edit Distance
47) EVAL    / Expected Number of Restriction Sites
48) MOTZ    / Motzkin Numbers and RNA Secondary Structures
49) NWCK    / Distances in Trees
50) SCSP    / Interleaving Two Motifs
51) SETO    / Introduction to Set Operations
52) SORT    / Sorting by Reversals
53) SPEC    / Inferring Protein from Spectrum
54) TRIE    / Introduction to Pattern Matching
55) CONV    / Comparing Spectra with the Spectral Convolution
56) CTBL    / Creating a Character Table
57) DBRU    / Constructing a De Bruijn Graph
58) EDTA    / Edit Distance Alignment
59) FULL    / Inferring Peptide from Full Spectrum
60) INDC    / Independent Segregation of Chromosomes
61) ITWV    / Finding Disjoint Motifs in a Gene
62) LREP    / Finding the Longest Multiple Repeat
63) NKEW    / Newick Format with Edge Weights
64) RNAS    / Wobble Bonding and RNA Secondary Structures
65) AFRQ    / Counting Disease Carriers
66) CSTR    / Creating a Character Table from Genetic Strings
67) CTEA    / Counting Optimal Alignments
68) CUNR    / Counting Unrooted Binary Trees
69) GLOB    / Global Alignment with Scoring Matrix
70) PCOV    / Genome Assembly with Perfect Coverage
71) PRSM    / Matching a Spectrum to a Protein
72) QRT     / Quartets
73) SGRA    / Using the Spectrum Graph to Infer Peptides
74) SUFF    / Encoding Suffix Trees
75) CHBP    / Character-Based Phylogeny
76) CNTQ    / Counting Quartets
77) EUBT    / Enumerating Unrooted Binary Trees
78) GASM    / Genome Assembly Using Reads
79) GCON    / Global Alignment with Constant Gap Penalty
80) LING    / Linguistic Complexity of a Genome
81) LOCA    / Local Alignment with Scoring Matrix
82) MEND    / Inferring Genotype from a Pedigree
83) MGAP    / Maximizing the Gap Symbols of an Optimal Alignment
84) MREP    / Identifying Maximal Repeats
85) MULT    / Multiple Alignment
86) PDPL    / Creating a Restriction Map
87) ROOT    / Counting Rooted Binary Trees
88) SEXL    / Sex-Linked Inheritance
89) SPTD    / Phylogeny Comparison with Split Distance
90) WFMD    / The Wright-Fisher Model of Genetic Drift
91) ALPH    / Alignment-Based Phylogeny
92) ASMQ    / Assessing Assembly Quality with N and N
93) CSET    / Fixing an Inconsistent Character Set
94) EBIN    / Wright-Fisher's Expected Behavior
95) FOUN    / The Founder Effect and Genetic Drift
96) GAFF    / Global Alignment with Scoring Matrix and Affine Gap Penalty
97) GREP    / Genome Assembly with Perfect Coverage and Repeats
98) OAP     / Overlap Alignment
99) QRTD    / Quartet Distance
100) SIMS   / Finding a Motif with Modifications
101) SMGB   / Semiglobal Alignment
102) KSIM   / Finding All Similar Motifs
103) LAFF   / Local Alignment with Affine Gap Penalty
104) OSYM   / Isolating Symbols in Alignments
105) RSUB   / Identifying Reversing Substitutions

*Note

I am using python3 ( 3.6.1 (v3.6.1:69c0db5, Mar 21 2017, 18:41:36) )
